### DISCUSSION TITLE
request: more nogc compatibility

### DISCUSSION THREAD

**Discussion by Moth-Tolias** [dlang/PhobosV3-Design#22] [Ideas]
Several useful features of phobos v2 allocate memory, making them unusable in performance-critical code. Some things simply can't be done safely without using the heap, of course, but it would be nice if phobos v3 was written with `@nogc` compatibility as a goal.

**Comment by LightBender** [comment]
That is the general direction that we're trying to go, but I want to make it clear that it's not going to be possible to completely remove the GC from Phobos.
https://github.com/LightBender/PhobosV3-Design/blob/main/Design.md#no-memory-allocation

**Reply by Moth-Tolias** [reply]
oh wow, somehow i forgot about that section when i originally read the document the other day and then completely blinded it when skimming through again today? how embarrassing - i'm terribly sorry!
